### PR TITLE
masterwork stat array

### PR DESCRIPTION
**Pull request by delphiactual**
#5364
MVP: Makes the array but only checks first value currently.

- [x] Make MW array
- [x] Use array elsewhere

@justrealmilk

### CODE REVIEW

File: src/app/inventory/store/masterwork.ts [hunk 1]
```diff
@@ -129,25 +129,20 @@ function buildMasterworkInfo(
   defs: D2ManifestDefinitions
 ): DimMasterwork | null {
   const socket = sockets.sockets.find((socket) => Boolean(socket.plug?.plugObjectives.length));
-  if (
-    !socket ||
-    !socket.plug ||
-    !socket.plug.plugObjectives ||
-    !socket.plug.plugObjectives.length
-  ) {
+  if (!socket?.plug?.plugObjectives?.length) {
```
Comment (delphiactual) [inline src/app/inventory/store/masterwork.ts:132]: I do not know if this is a valid change but I believe it is.

Comment (justrealmilk) [reply src/app/inventory/store/masterwork.ts:132]: i welcome this

File: src/app/inventory/store/masterwork.ts [hunk 2]
```diff
@@ -110,12 +110,12 @@ function buildForsakenMasterworkStats(
...
-      statName: statDef.displayProperties.name,
-      statValue: masterworkSocket.plug.stats
-        ? masterworkSocket.plug.stats[masterwork.statTypeHash]
-        : 0,
       tier: masterwork.value,
+      stats: {
+        hash: masterwork.statTypeHash,
+        name: statDef.displayProperties.name,
+        value: masterworkSocket.plug.stats?.[masterwork.statTypeHash] || 0,
+      }[0],
```
Comment (sundevour) [inline src/app/inventory/store/masterwork.ts:118]: i don'y think this does what you think it does unless i am really, really confused on syntax

Comment (delphiactual) [reply src/app/inventory/store/masterwork.ts:118]: it puts that into index 0? without it it causes a type error.

Comment (sundevour) [reply src/app/inventory/store/masterwork.ts:118]: you're creating an object then immediately selecting index 0 from it (which it doesn't have)
image

i think what's intended here is 
```
      stats: [{
        hash: masterwork.statTypeHash,
        name: statDef.displayProperties.name,
        value: masterworkSocket.plug.stats?.[masterwork.statTypeHash] || 0,
      }],
```

**Comment by delphiactual** [comment]
all that remains is synergy stat, columns, and spreadsheet... all of which require an extra row...

**Comment by delphiactual** [comment]
I functionized `getMasterworkStatNames` but I may have put it into the wrong `util` helper. Let me know if I need to move it

File: src/app/inventory/store/masterwork.ts (hunk 1 above)
Comment (bhollis) [reply src/app/inventory/store/masterwork.ts:132]: yup that works

**Pull request merged**